New deployment scenarios (#3666)
zenml-io/zenml: docs/book/getting-started/system-architectures.md

</details>

### ZenML Pro partially Self-Hosted Architecture

![ZenML Pro self-hosted deployment](../.gitbook/assets/cloud_architecture_scenario_1_2.png)

> The partially self-hosted architecture offers a balanced approach that combines the benefits of cloud-hosted control with on-premises data sovereignty. In this configuration, while the ZenML Pro control plane remains hosted by ZenML (handling user management, authentication, and global workspace coordination), all other components - including services, data, and secrets - are deployed within your own cloud infrastructure.

Reviewer: ```suggestion
The partially self-hosted architecture offers a balanced approach that combines the benefits of cloud-hosted control with on-premises data sovereignty. In this configuration, while the ZenML Pro control plane remains hosted by ZenML (handling user management, authentication, RBAC and global workspace coordination), all other components - including services, data, and secrets - are deployed within your own cloud infrastructure.
```

> This hybrid model is particularly well-suited for organizations with:
> * A centralized MLOps or Platform team responsible for standardizing ML practices
> * Multiple business units or teams that require autonomy over their data and infrastructure
> * Compliance requirements that mandate keeping sensitive data and ML artifact metadata within company infrastructure
> * Need for customization of workspace configurations while maintaining centralized governance
> 
> The key advantages of this setup include:
> * Simplified user management through the ZenML-hosted control plane
> * Complete control over sensitive data and ML artifacts
> * Ability to customize and configure workspaces according to specific team needs
> * Reduced operational overhead compared to fully self-hosted deployments
> * Easier updates and maintenance of the control plane components
> 

Reviewer: ```suggestion
This hybrid model is particularly well-suited for organizations with:
* A centralized MLOps or Platform team responsible for standardizing ML practices
* Multiple business units or teams that require autonomy over their data and infrastructure
* Strict security requirements where workspaces must operate behind VPN/corporate firewalls
* Compliance requirements that mandate keeping sensitive data and ML artifact metadata within company infrastructure
* Need for customization of workspace configurations while maintaining centralized governance

The key advantages of this setup include:
* Simplified user management through the ZenML-hosted control plane
* Complete data sovereignty - sensitive data and ML artifacts remain within your infrastructure
* Secure networking - workspaces communicate through outbound-only connections via VPN/private networks
* Ability to customize and configure workspaces according to specific team needs
* Reduced operational overhead compared to fully self-hosted deployments
* Reduced maintenance burden - all control plane updates and maintenance are handled by ZenML
```

| **ZenML Pro SaaS (Customer Secret Store)** | ML metadata in ZenML-hosted DB; Actual ML data artifacts in customer Artifact Store; Secrets in customer-managed Secret Store | Metadata flows to ZenML Pro Control Plane; ML data artifacts stay on customer infrastructure; All credential access requires going through customer Secret Store | ZenML Pro must request credentials from customer Secret Store; Additional security layer for infrastructure access; No ML data stored on ZenML infrastructure | Improved security posture; customer maintains control of all credentials while benefiting from ZenML Pro features |

> | **ZenML Pro Partially Self-Hosted** | Control Plane on ZenML infrastructure; Workspace, DB, Secret Store, Orchestrator, and Artifact Store on customer infrastructure | Only authentication/authorization data flows to ZenML; All ML data and metadata stays on customer infrastructure | ZenML Control Plane has limited access to user management data; No access to actual ML data or metadata; Customer maintains all data access controls | Strong data isolation with only authentication events crossing boundary |

Reviewer: ```suggestion
| **ZenML Pro Partially Self-Hosted** | Control Plane on ZenML infrastructure; Workspace, DB, Secret Store, Orchestrator, and Artifact Store on customer infrastructure | Only authentication/authorization data flows to ZenML; All ML data and metadata stays on customer infrastructure | ZenML Control Plane has limited access to user management data; No access to actual ML data or metadata; Customer maintains all data access controls | Strong data isolation with only authentication events crossing boundary. Allows securing access via VPN/private networks. |
```
